scripts/core/ui/components/video.tsx

```diff
@@ -19,12 +21,18 @@ export class VideoComponent extends React.PureComponent<IProps> {
         }
 
         const videoRenditions = Object.values(item.renditions)
-            .filter((rendition) => rendition.mimetype != null && rendition.mimetype.startsWith('video'));
+            .filter(
+                (rend) => rend.mimetype != null && (rend.mimetype.startsWith('video') || rend.mimetype.startsWith('application/x-mpegurl'))
+            );
+
+        if (videoRenditions.length > 0 && videoRenditions[0].mimetype === 'application/x-mpegurl') {
+            return <HLSVideoComponent poster={poster} streamUrl={videoRenditions[0].href}/>
```

Comment:
If you add a `key` when rendering a component and that key changes, the component will fully remount and trigger all the usual lifecycle methods including `componentWillUnmount` and `componentDidMount`. If you set `videoRenditions[0].href` as a key, you can get rid off `HLSVideoComponent.componentDidUpdate` method.